pep-9999.rst

```diff
@@ -0,0 +1,883 @@
+and documented in several other system/distro and application package formats is
+provided as an informative Appendix in this PEP.
+
+There are a few takeaways from this survey:
+
+- Most package formats use a single `license` field.
+
+- Many modern package formats use some form of license expression syntax to
+  optionally combine more than one license identifiers together. SPDX and
+  SPDX-like syntaxes are the most popular in use.
+
+- SPDX license ids are becoming a de-facto way to reference common licenses
+  everywhere, whether or not a license expression syntax is used. But they often
+  need to be supplemented with extra license ids or conventions to accept extra
+  or generic licenses such as "Proprietary" or "Public domain" not tracked by
+  SPDX.
+
+- Several package format support documenting both a license expression and
+  the paths of the corresponding files that contain the license text. Most free
+  and open source software licenses require to include their full text in a
+  distribution.
+
+These considerations have guided the design and updates proposed in this PEP.
+
+Reusing the existing `license` field with license expressions will provide a an
+intuitive and more structured way to express the license of a distribution using
+a well defined syntax and well known license ids.
+
+Over time, recommending the usage of these expressions will help Python package
+publishers improve the clarity of their license documentation to the benefit of
+packages authors, consumers and redistributors.
+
+
+Core Metadata Specification updates
+===================================
+
+The canonical source for the names and semantics of each of the supported
+metadata fields is the Core Metadata Specification [#cms]_ document.
+
+The details of the updates considered to the Core Metadata Specification [#cms]_
+document as part of his PEP are detailed here and will be added to the
+canonical source once this PEP is approved.
+
+
+Added in Version 2.2
+--------------------
+
+License-File (multiple use)
+:::::::::::::::::::::::::::
+
+The is a string that is a package-root relative path to a license file. The
+license file content __must__ be UTF-8-encoded text.
+
+Build tools should honor this field and include the corresponding license
+file(s) in the built package.
+
+
+Changed in Version 2.2
+----------------------
+
+License (optional)
+::::::::::::::::::
+
+Text indicating the license covering the distribution. This text can be either a
+valid `License Expression` as defined here or any free text.
+
+Publishing tools SHOULD issue an informational warning if this field is empty or
+missing or is not a valid `License Expression` as defined here. Build tools MAY
+issue such a warning too.
+
+
+License Expression syntax
+'''''''''''''''''''''''''
+
+A `License Expression` is a string using the SPDX license expression syntax as
+documented in the SPDX specification [#spdx]_ using either Version 2.1
```

Comment:
I'd explicitly discourage use of `+` combinator. Recent license lists added `GPL-2.0-or-later` and `GPL-2.0-only` identifiers, and deprecated `GPL-2.0`. I don't remember proper rationale, but that change removed most needs for `+`.

Reply:
I stand corrected: https://spdx.org/ids-how it's an issue with FSF license only.

Reply:
The `+` is mostly abandoned at this stage and the main area where this was showing up was for GNU licenses and as you rightly pointed out SPDX changed these to `-only` and `or-later` based on requests from FSF.

> I don't remember proper rationale, but that change removed most needs for +.

That was a request from rms and the FSF. See https://www.gnu.org/licenses/identify-licenses-clearly.html and then https://www.fsf.org/blogs/rms/rms-article-for-claritys-sake-please-dont-say-licensed-under-gnu-gpl-2